airbyte-integrations/bases/bases-destination-jdbc/src/main/java/io/airbyte/integrations/destination/staging/AsyncFlush.java

```diff
@@ -93,8 +93,12 @@ public void flush(final StreamDescriptor decs, final Stream<PartialAirbyteMessag
 
   @Override
   public long getOptimalBatchSizeBytes() {
-    // todo(davin): this should be per-destination specific. currently this is for Snowflake.
-    return 100 * 1024 * 1024;
+    // todo(ryankfu): this should be per-destination specific. currently this is for Snowflake.
+    // The size chosen is currently for improving the performance of low memory connectors. With 1 Gi of
+    // resource the connector will usually at most fill up around 150 MB in a single queue. By lowering
+    // the batch size, the AsyncFlusher will flush in smaller batches which allows for memory to be
+    // freed earlier similar to a sliding window effect
+    return 50 * 1024 * 1024;
```

Comment:
[non blocking] it might be interesting to make the batch size more dynamic, i.e. if we can introspect the allocated memory and set the batch size to be x% of that memory

Reply:
Highly agree, I'll file out a ticket since the scope of this work is to fix bugs but it's worthwhile if performance can be a dynamic factor